src/NewEditor/index.tsx

```diff
@@ -158,13 +176,23 @@ const Canvas: React.FC<EditorProps> = ({ nodes }) => {
 
   const containerRef = useCallback((element) => {
     if (element !== null) {
+      const rect = element.getBoundingClientRect()
+
       setOffset({
-        offsetLeft: element.offsetLeft,
-        offsetTop: element.offsetTop
+        offsetLeft: rect?.left || 0,
+        offsetTop: rect?.top || 0
       })
     }
   }, [])
 
+  useEffect(() => {
+    if (!dotSize && stateNodes.length) {
+      const rect = document.querySelector(".dot.input")?.getBoundingClientRect()
```

Comment:
Not good, maybe we can use id for this object. Because if we will rename classname, we forget about this.